pkg/resources/mixerlesstelemetry/mixerlesstelemetry.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/banzaicloud/istio-operator/pkg/k8sutil"
+	"github.com/banzaicloud/istio-operator/pkg/resources"
+	"github.com/banzaicloud/istio-operator/pkg/util"
+)
+
+const (
+	componentName = "mixerless-telemetry"
+)
+
+type Reconciler struct {
+	resources.Reconciler
+	dynamic dynamic.Interface
+}
+
+func New(client client.Client, dc dynamic.Interface, config *istiov1beta1.Istio) *Reconciler {
+	return &Reconciler{
+		Reconciler: resources.Reconciler{
+			Client: client,
+			Config: config,
+		},
+		dynamic: dc,
+	}
+}
+
+func (r *Reconciler) Reconcile(log logr.Logger) error {
+	log = log.WithValues("component", componentName)
+
+	log.Info("Reconciling")
+
+	desiredState := k8sutil.DesiredStateAbsent
+	if util.PointerToBool(r.Config.Spec.MixerlessTelemetry.Enabled) {
+		desiredState = k8sutil.DesiredStatePresent
+	}
+
+	drs := []resources.DynamicResourceWithDesiredState{
+		{DynamicResource: r.metaexchangeEnvoyFilter},
+		{DynamicResource: r.statsEnvoyFilter},
+	}
+
+	for _, dr := range drs {
+		o := dr.DynamicResource()
+		err := o.Reconcile(log, r.dynamic, desiredState)
+		if err != nil {
+			return emperror.WrapWith(err, "failed to reconcile dynamic resource", "resource", o.Gvr)
+		}
+	}
+
+	log.Info("Reconciled")
+
+	return nil
+}
+
+func (r *Reconciler) metaexchangeEnvoyFilter() *k8sutil.DynamicObject {
+	filterInbound := map[string]interface{}{
+		"filterConfig": map[string]interface{}{
+			"configuration": "envoy.wasm.metadata_exchange",
+			"vm_config": map[string]interface{}{
+				"code": map[string]interface{}{
+					"inline_string": "envoy.wasm.metadata_exchange",
+				},
+				"vm": "envoy.wasm.vm.null",
+			},
+		},
+		"filterName": "envoy.wasm",
+		"filterType": "HTTP",
+		"insertPosition": map[string]interface{}{
+			"index": "FIRST",
+		},
+		"listenerMatch": map[string]interface{}{
+			"listenerProtocol": "HTTP",
+			"listenerType":     "SIDECAR_INBOUND",
+		},
+	}
+
+	var filterOutbound map[string]interface{}
+	filterOutbound, _ = util.Map(filterInbound)
+	filterOutbound["listenerMatch"].(map[string]interface{})["listenerType"] = "SIDECAR_OUTBOUND"
+
+	var filterGateway map[string]interface{}
+	filterGateway, _ = util.Map(filterInbound)
+	filterGateway["listenerMatch"].(map[string]interface{})["listenerType"] = "GATEWAY"
+
+	return &k8sutil.DynamicObject{
+		Gvr: schema.GroupVersionResource{
+			Group:    "networking.istio.io",
+			Version:  "v1alpha3",
+			Resource: "envoyfilters",
+		},
+		Kind:      "EnvoyFilter",
+		Name:      componentName + "-metadata-exchange",
+		Namespace: r.Config.Namespace,
+		Spec: map[string]interface{}{
+			"filters": []map[string]interface{}{
+				filterInbound,
+				filterOutbound,
+				filterGateway,
+			},
+		},
+		Owner: r.Config,
+	}
+}
+
+func (r *Reconciler) statsEnvoyFilter() *k8sutil.DynamicObject {
+	filterInbound := map[string]interface{}{
+		"filterConfig": map[string]interface{}{
+			"configuration": heredoc.Doc(`
+{
```

Comment:
I beleive this could be indented like it was a code block, heredoc should remove the obsolete whitespaces

Reply:
👍